core/src/main/java/com/sequenceiq/cloudbreak/service/decorator/StackDecorator.java

```diff
@@ -108,6 +110,13 @@ public Stack decorate(@Nonnull Stack subject, @Nonnull StackV4Request request, U
         LOGGER.debug("Domain was prepared under {} ms for stack {}", System.currentTimeMillis() - start, stackName);
 
         String credentialName = request.getEnvironment().getCredentialName();
+        if (Objects.isNull(credentialName)) {
+            Set<EnvironmentView> environments = environmentViewService.findByNamesInWorkspace(Set.of(request.getEnvironment().getName()), workspace.getId());
+            if (!CollectionUtils.isEmpty(environments)) {
```

Comment:
I think it is better: 
`environmentViewService.getByNameForWorkspace(request.getEnvironment().getName(), workspace)`